### PR TITLE
configure: apply autoupdate to reflect new autoconf syntax

### PR DESCRIPTION
**Pull request by norg**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

https://redmine.openinfosecfoundation.org/issues/879

Describe changes:
This updates configure.ac to the newest autoconf syntax. It's tested on OISF Buildbot, ArchLinux and Ubuntu 16.04 LTS.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

- PR norg-pcap: https://buildbot.openinfosecfoundation.org/builders/norg-pcap/builds/56
- PR norg: https://buildbot.openinfosecfoundation.org/builders/norg/builds/58

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -269,11 +266,8 @@
     AS_IF([test "x$enable_tls" = "xyes"], [
         # check if our target supports thread local storage
         AC_MSG_CHECKING(for thread local storage __thread support)
-        AC_TRY_COMPILE([#include <stdlib.h>],
-            [ static __thread int i; i = 1; i++; ],
-            [AC_DEFINE([TLS], [1], [Thread local storage])
-             AC_MSG_RESULT([yes]) ],
-            [AC_MSG_RESULT([no])])
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <stdlib.h>]], [[ static __thread int i; i = 1; i++; ]])],[AC_DEFINE([TLS], [1], [Thread local storage])
```
Comment (victorjulien) [inline configure.ac]: this is a bit odd, the old code is much more readible

File: configure.ac [hunk 2]
```diff
@@ -2152,7 +2133,8 @@ AC_SUBST(CONFIGURE_SYSCONDIR)
 AC_SUBST(CONFIGURE_LOCALSTATEDIR)
 AC_SUBST(PACKAGE_VERSION)
 
-AC_OUTPUT(Makefile src/Makefile rust/Makefile rust/Cargo.toml rust/.cargo/config qa/Makefile qa/coccinelle/Makefile rules/Makefile doc/Makefile doc/userguide/Makefile contrib/Makefile contrib/file_processor/Makefile contrib/file_processor/Action/Makefile contrib/file_processor/Processor/Makefile contrib/tile_pcie_logd/Makefile suricata.yaml scripts/Makefile scripts/suricatasc/Makefile scripts/suricatasc/suricatasc etc/Makefile etc/suricata.logrotate etc/suricata.service)
+AC_CONFIG_FILES([Makefile src/Makefile rust/Makefile rust/Cargo.toml rust/.cargo/config qa/Makefile qa/coccinelle/Makefile rules/Makefile doc/Makefile doc/userguide/Makefile contrib/Makefile contrib/file_processor/Makefile contrib/file_processor/Action/Makefile contrib/file_processor/Processor/Makefile contrib/tile_pcie_logd/Makefile suricata.yaml scripts/Makefile scripts/suricatasc/Makefile scripts/suricatasc/suricatasc etc/Makefile etc/suricata.logrotate etc/suricata.service])
```
Comment (victorjulien) [inline configure.ac]: it seems we can have multiple AC_CONFIG_FILES instances. That would be a lot nicer than this large line

**Comment by norg** [comment]
The main issue is, that we might want to add some check within our buildbot/CI to check if commits/PRs don't use the old format.

I would also need to do it again, since some changes were made since the last time I ran autoupdate. So IMHO the best thing would be to use a closed merge window, then apply autoupdate, fix parts like you mentioned above and from that point on force the new syntax for ne PRs.

Maybe @regit has more thoughts about that.

**Comment by victorjulien** [comment]
Think the CI question is valid, but also a bit separate from this PR. So if you can resubmit we can get the changes in and continue to think about how to check for it in CI.

**Pull request closed without merging**